### PR TITLE
eosiocpp link

### PR DESCRIPTION
**Pull request by lyjstudy**
ubuntu16.4 64bit only link one file

### CODE REVIEW

**Comment by spoonincode** [comment]
Can you explain what this is intended to fix or solve?

**Comment by lyjstudy** [comment]
WASM_LLVM_LINK does not support wildcards

**Comment by lyjstudy** [comment]
a.cpp:
```
#include <eosiolib/eosio.hpp>
int add(int a, int b);
using namespace eosio;
class test : public contract {
    public:
        using contract::contract;
        /// @abi action
        void show(account_name who) {
            require_auth( who );
            print(add(1, 2));
        }
};
EOSIO_ABI( test, (show) )
```
a1.cpp
```
int add(int a, int b) {
    return a + b;
}
```
command:
```
eosiocpp -o a.wast a.cpp a1.cpp
```

**Comment by lyjstudy** [comment]
llvm-link dependent input order, May be LLVM BUG,
command: when first letter is the same
```
eosiocpp -o a.wast a.cpp ab.cpp
```

**Comment by spoonincode** [comment]
ah yes, you are right, there is a dependency order on llvm-link's parameters. We already got burned by this limitation another way (PR #2737).

File: tools/eosiocpp.in [hunk 1]
```diff
@@ -51,7 +51,12 @@ function build_contract {
 
     done
 
-    ($PRINT_CMDS; @WASM_LLVM_LINK@ -only-needed -o $workdir/linked.bc $workdir/built/* \
+    linkfiles=''
+    for file in `ls -t $workdir/built/ | sort -r`; do
```
Comment (spoonincode) [inline tools/eosiocpp.in]: This isn't fixing the underlying issue. Your original fix, where you construct `$linkfiles` ordered by how they are given on command line, is preferable I believe.

**Comment by lyjstudy** [comment]
C library and C++ library should not be used, unless it is headeronly.

**Comment by lyjstudy** [comment]
sorry, maybe llvm-link -only-needed rule?.
first file is major file

**Comment by lyjstudy** [comment]
Not the best solution

**Pull request closed without merging**